api/src/main/java/uk/gov/dhsc/htbhf/claimant/scheduler/NewCardScheduleService.java

```diff
@@ -0,0 +1,28 @@
+package uk.gov.dhsc.htbhf.claimant.scheduler;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.dhsc.htbhf.claimant.entity.Claim;
+import uk.gov.dhsc.htbhf.claimant.repository.ClaimRepository;
+
+import java.util.stream.Stream;
+
+@Service
+@AllArgsConstructor
+@Slf4j
+public class NewCardScheduleService {
+
+    private ClaimRepository claimRepository;
+
+    /**
+     * Print out first 10 new claims (testing purposes only).
+     */
+    @Transactional
+    public void createNewCards() {
+        Stream<Claim> newClaims = claimRepository.getNewClaims();
+        newClaims.limit(10).forEach(claim ->
```

Comment:
I think a useful bit of logging here would be to log out at info level the total number of new claims found.

Reply:
Won't that be quite inefficient as it'll will require fetching all the data for the count, then once again for processing.

Reply:
you could however keep a count and log it at the end

Reply:
Mind you, this method will need to change quite substantially. We will need each new card to be in its own transaction, so the method shouldn't be `@Transactional` - rather it should invoke a component to perform the task for a single claim and _that method_ is `@Transactional`

Reply:
Yes, the structure of it will change greatly, once it has solidified a bit more we should make sure we do log a count, but we should be able to do this using the job's metadata on how many items it is processing, so we don't need to query the data twice.

Reply:
The method has to be transactional, spring jpa won't allow a repository returning a stream to be called from the method otherwise.

Reply:
Moved iterating over new claims to job.

Reply:
I don't see any change here, what did you do?